src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +17,34 @@
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+        } catch (IllegalArgumentException e) {
+            throw new DataProcessingException("The instance of Movie is not an entity.", e);
+        } catch (IllegalStateException | PersistenceException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot save the movie: " + movie, e);
```

Comment:
The exception message could be more descriptive by including more context about the movie entity, such as its title or other identifying attributes, to aid in debugging.